internal/trust/remotes.go

```diff
@@ -68,9 +68,15 @@ func (r *Remotes) Load(dir string) error {
 			return fmt.Errorf("Failed to parse local record %q. Found empty certificate", remote.Name)
 		}
 
-		r.data[remote.Name] = *remote
+		remoteData[remote.Name] = *remote
 	}
 
+	if len(remoteData) == 0 {
+		return fmt.Errorf("Failed to parse new remotes from truststore")
+	}
+
+	r.data = remoteData
```

Comment:
Do we need a mutex around this?

Reply:
See [below](https://github.com/canonical/microcluster/pull/62#discussion_r1204566152)